pkg/nsx/services/subnet/subnet.go

```diff
@@ -352,26 +342,56 @@ func (service *SubnetService) GetSubnetByPath(path string) (*model.VpcSubnet, er
 	return nsxSubnet, err
 }
 
-func (service *SubnetService) ListSubnetID() sets.Set[string] {
+func (service *SubnetService) ListSubnetSetIDsFromNSXSubnets() sets.Set[string] {
+	subnetSetIDs := service.SubnetStore.ListIndexFuncValues(common.TagScopeSubnetSetCRUID)
+	return subnetSetIDs
+}
+
+func (service *SubnetService) ListSubnetIDsFromNSXSubnets() sets.Set[string] {
+	subnetIDs := service.SubnetStore.ListIndexFuncValues(common.TagScopeSubnetCRUID)
+	return subnetIDs
+}
+
+// ListIndexFuncValues returns all the indexed values of the given index
+// // Index maps the indexed value to a set of keys in the store that match on that value
+// type Index map[string]sets.String
+//
+//	func (i *storeIndex) getIndexValues(indexName string) []string {
+//		index := i.indices[indexName]
+//		names := make([]string, 0, len(index))
+//		for key := range index {
+//			names = append(names, key)
+//		}
+//		return names
+//	}
```

Comment:
stale code?

Reply:
I think it's not easy to understand here, so added some more annotations here.